js/site.js

```diff
@@ -82,6 +86,15 @@ changeset_info.innerHTML = '<div class="loading">loading...</div>';
 
 var lastLocation = L.latLng(0, 0);
 
+function isBboxSizeAcceptable(bbox) { // heuristic to fetch
+    var width = Math.abs(bbox.getSouthWest().lat - bbox.getNorthEast().lat);
+    var height = Math.abs(bbox.getSouthWest().lng - bbox.getNorthEast().lng);
+    // A guesstimate of the maximum filtered area size that the server would accept.
+    // For larger areas, we fall back to the global (server-cached) change file
+    // ...and we process it client-side as usual.
+    return (width * height) < 2;
```

Comment:
This is only enforced as a hard limit here - e.g. "3" also kind of works - but I don't want to load the Overpass server too much